pkg/template/controller/templateinstance_controller.go

```diff
@@ -93,32 +109,77 @@ func (c *TemplateInstanceController) handle(templateInstance *templateapi.Templa
 		}
 	}
 
-	_, err = c.templateclient.TemplateInstances(templateInstance.Namespace).Update(templateInstance)
+	_, err = c.templateclient.TemplateInstances(templateInstance.Namespace).UpdateStatus(templateInstance)
 	if err != nil {
 		utilruntime.HandleError(fmt.Errorf("TemplateInstance update failed: %v", err))
 	}
 	return err
```

Comment:
(or rather than retrying the key, we need to enqueue a different event that just tells the controller to redo the update action, not the full provision+update action. @csrwng has some experience w/ this)